zerver/tests/test_message_send.py

```diff
@@ -2364,6 +2364,33 @@ def test_get_raw_unread_data_for_group_direct_messages(self) -> None:
         )
         self.assertEqual(recent_conversation["max_message_id"], message2_id)
 
+    def test_stream_becomes_active_on_message_send(self) -> None:
+        # Mark a stream as inactive
+        stream = self.make_stream("inactive_stream")
+        stream.is_recently_active = False
+        stream.save()
+        self.assertEqual(stream.is_recently_active, False)
+
+        # Send a message to the stream
+        sender = self.example_user("hamlet")
+        self.subscribe(sender, stream.name)
+        sending_client = make_client(name="test suite")
+        message = Message(
+            sender=sender,
+            recipient=stream.recipient,
+            realm=stream.realm,
+            content="message_content",
+            date_sent=timezone_now(),
+            sending_client=sending_client,
+        )
+        message.set_topic_name("topic_name")
+        message_dict = build_message_send_dict(message=message)
+        do_send_messages([message_dict])
```

Comment:
Can't you do just `self.send_stream_message`? Usually we use those test class helpers are better than building a fake `Message` like this, which is fragile (both the exact steps, like making a client being code we have to maintain, and also not testing the actual production code path).

Reply:
Done